src/haptics.cpp

```diff
@@ -2,27 +2,36 @@
 
 void
 init_haptics() {
-    // Initialize Motor Pins
-    for (int i = 0; i < N_MOTORS; i++) {
-        // Setup EN pin and set to low
-        pinMode(MOTOR_PINS[i] + EN_OFFSET, OUTPUT);
-        digitalWrite(MOTOR_PINS[i] + EN_OFFSET, LOW);
-        pinMode(MOTOR_PINS[i], OUTPUT);
-    }
+  // Initialize Motor Pins
+  for (int i = 0; i < N_MOTORS; i++) {
+    // Setup EN pin and set to low
+    pinMode(MOTOR_PINS[i] + EN_OFFSET, OUTPUT);
+    digitalWrite(MOTOR_PINS[i] + EN_OFFSET, LOW);
+    pinMode(MOTOR_PINS[i], OUTPUT);
+  }
+}
+
+void actuate_motor(int motor_pin, int duration, int frequency, double percent_motor) {
```

Comment:
duration: notice that that the loop on line 19 runs each duration for 2*frequency milliseconds. Maybe have duration be specified in milliseconds (instead of motor vibration cycles) and have a line of math to calculate how many times the loop should run.

Reply:
Motor frequency is also unchanging for this particular motor we are using, so can be a constant in the function that doesn't have to be passed in (like how the 85% maximum drive is also a motor-specific constant).

Reply:
Ok, so would the frequency constant be 170?

Reply:
It stays at 3, since that's the amount of time in milliseconds for half of a cycle. 1/(.006s) = 166.66Hz, which is close enough to 170Hz. Here's a illustration of what we're doing with the analogWrites to simulate a sinusoid at 167Hz. (In our case, unlike in the illustration, the duty cycle doesn't change during those 3ms half-cycles.)

![proxy-image(8)sd](https://user-images.githubusercontent.com/16432322/150926376-03e08971-45af-415a-ad78-b850230a37fc.png)

Reply:
But you could say that you want to specify the constant as 170Hz, then have some math to turn that into the half-period time: 1/(170Hz) / 2 = 0.00294s. You could get closer to this value than before by multiplying by 10^6 to get the number of microseconds to delay, and use `delayMicroseconds` instead of `delay`

Reply:
Also sorry about how not obvious all of this is. I'll admit this PWM scheme to simulate the sinusoid is a bit funny. The motor parameters come from the datasheet which can be found [here](https://www.digikey.com/en/products/detail/jinlong-machinery-electronics-inc/G1040001D/10285887).

Reply:
No worries, that makes sense. Thanks!